api/resolvers/paidAction.js

```diff
@@ -50,35 +50,44 @@ export default {
     }
   },
   Mutation: {
-    retryPaidAction: async (parent, { invoiceId }, { models, me, lnd }) => {
+    retryPaidAction: async (parent, { invoiceId, newAttempt }, { models, me, lnd }) => {
       if (!me) {
         throw new Error('You must be logged in')
       }
 
-      const invoice = await models.invoice.findUnique({ where: { id: invoiceId, userId: me.id } })
+      // make sure only one client at a time can retry by immediately transitioning to an intermediate state
+      const [invoice] = await models.$queryRaw`
+          UPDATE "Invoice"
+          SET "actionState" = 'RETRY_PENDING'
+          WHERE id in (
+            SELECT id FROM "Invoice"
+            WHERE id = ${invoiceId} AND "userId" = ${me.id} AND "actionState" = 'FAILED'
+            FOR UPDATE
+          )
+          RETURNING *`
       if (!invoice) {
         throw new Error('Invoice not found')
       }
 
-      if (invoice.actionState !== 'FAILED') {
-        if (invoice.actionState === 'PAID') {
-          throw new Error('Invoice is already paid')
-        }
-        throw new Error(`Invoice is not in failed state: ${invoice.actionState}`)
```

Comment:
removing these error messages probably makes error messages during bugs worse, for example if `retry` via notifications fail

Reply:
I think the thing to do in this case is return the pending/paid invoice rather than an error.

We should do the same thing in #1669 - if an attempt is made to vote again, just return the original vote.

It's kind of painting over the bug - which is some kind of clientside cache issue - but it's way better UX.

Reply:
wouldn't this cause the client to retry a paid invoice if the retry mutation doesn't throw?

Reply:
Yeah, we'd have to modify the client logic when retrying, or ....

we could just throw a more specific error, catch it, then update the cache appropriately.

Reply:
I have put this PR back into review after some Q&A of 89a6b42. I think what we discuss here should be done in a separate PR.

Reply:
Oh for sure. I was just riffing on your comment. I keep encountering that poll bug.